tests/phpunit/integration/Core/User_Input/User_InputTest.php

```diff
@@ -197,4 +207,75 @@ public function test_set_answers() {
 			$response
 		);
 	}
+
+	public function test_set_answers__preserves_purpose_answer_user_attribution() {
+		$second_admin_id = $this->factory()->user->create( array( 'role' => 'administrator' ) );
+
+		$this->user_input->set_answers(
+			array(
+				'purpose'       => array( 'publish_blog' ),
+				'postFrequency' => array( 'daily' ),
+				'goals'         => array( 'growing_audience', 'finding_new_topics' ),
+			)
+		);
+
+		$existing_answers = $this->user_input->get_answers();
+
+		$this->assertEquals( $existing_answers['purpose']['answeredBy'], $this->user_id );
+
+		$this->user_options->switch_user( $second_admin_id );
+
+		$answers = array(
+			'purpose'       => array( 'publish_blog' ),
+			'postFrequency' => array( 'weekly' ),
+			'goals'         => array( 'improving_performance' ),
+		);
+		$this->user_input->set_answers( $answers );
+
+		$existing_answers = $this->user_input->get_answers();
+		// Since purpose answer didn't change, it should still be attributed to admin 1.
+		$this->assertEquals( $existing_answers['purpose']['answeredBy'], $this->user_id );
+
+		$this->user_input->set_answers( $answers );
+		$existing_answers = $this->user_input->get_answers();
+		// Since no answer changed, purposr answer should still be attributed to admin 1.
+		$this->assertEquals( $existing_answers['purpose']['answeredBy'], $this->user_id );
```

Comment:
```suggestion
		$this->user_input->set_answers( $answers );
		$existing_answers = $this->user_input->get_answers();
		// Since no answer changed, the "purpose" answer should still be attributed to admin 1.
		$this->assertEquals( $existing_answers['purpose']['answeredBy'], $this->user_id );
```

Let's move this to its own test spec/function, just for clarity. I know it's technically testing the behaviour here, but it's a bit hard to follow the logic of this here. It's probably easier if we just have a new test spec that sets the same answers separate to this one 🙂

Reply:
Makes sense, thanks.

PR Updated